src/System.IO.FileSystem/tests/Directory/Exists.cs

```diff
@@ -1,382 +1,287 @@
 
-            Assert.False(result, path);
+            Assert.False(Exists(IOServices.RemoveTrailingSlash(path)));
+            Assert.False(Exists(IOServices.RemoveTrailingSlash(IOServices.RemoveTrailingSlash(path))));
+            Assert.False(Exists(IOServices.RemoveTrailingSlash(IOServices.AddTrailingSlashIfNeeded(path))));
         }
-    }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // alternate data stream
-    public static void Exists_PathWithAlternativeDataStreams_ReturnsFalse()
-    {
-        var paths = IOInputs.GetPathsWithAlternativeDataStreams();
-        foreach (var path in paths)
+        [Fact]
+        public void PathAlreadyExistsAsDirectory()
         {
-            bool result = Directory.Exists(path);
+            string path = GetTestFilePath();
+            DirectoryInfo testDir = Directory.CreateDirectory(path);
 
-            Assert.False(result, path);
+            Assert.True(Exists(IOServices.RemoveTrailingSlash(path)));
+            Assert.True(Exists(IOServices.RemoveTrailingSlash(IOServices.RemoveTrailingSlash(path))));
+            Assert.True(Exists(IOServices.RemoveTrailingSlash(IOServices.AddTrailingSlashIfNeeded(path))));
         }
-    }
 
-    [Fact]
-    [OuterLoop]
-    [PlatformSpecific(PlatformID.Windows)] // device names
-    public static void Exists_PathWithReservedDeviceNameAsPath_ReturnsFalse()
-    {
-        var paths = IOInputs.GetPathsWithReservedDeviceNames();
-        foreach (var path in paths)
+        [Fact]
+        public void DotAsPath_ReturnsTrue()
         {
-            bool result = Directory.Exists(path);
-
-            Assert.False(result, path);
+            Assert.True(Exists(Path.Combine(TestDirectory, ".")));
         }
-    }
-
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // UNC paths
-    public static void Exists_UncPathWithoutShareNameAsPath_ReturnsFalse()
-    {
-        var paths = IOInputs.GetUncPathsWithoutShareName();
 
-        foreach (string path in paths)
+        [Fact]
+        public void DirectoryGetCurrentDirectoryAsPath_ReturnsTrue()
         {
-            bool result = Directory.Exists(path);
-
-            Assert.False(result, path);
+            Assert.True(Exists(Directory.GetCurrentDirectory()));
         }
-    }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // max directory length not fixed on Unix
-    public static void Exists_DirectoryEqualToMaxDirectory_ReturnsTrue()
-    {   // Creates directories up to the maximum directory length all at once
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        public void DotDotAsPath_ReturnsTrue()
         {
-            PathInfo path = IOServices.GetPath(directory.Path, IOInputs.MaxDirectory, maxComponent: 10);
+            Assert.True(Exists(Path.Combine(TestDirectory, GetTestFileName(), "..")));
+        }
 
-            Directory.CreateDirectory(path.FullPath);
+#if !TEST_WINRT // WinRT cannot access root
+        /*
```

Comment:
In addition to being conditionally compiled in, this is also commented out.  Do we still need this?  Are we missing cases from being tested because this is commented out?